### PR TITLE
Updated Vega-Embed on the site

### PR DESCRIPTION
**Pull request by yhoonkim**
Updated some pages of the site to use new Vega-Embed. 

My major concern of this PR is about `streaming.ts`. I changed the code of it under an assumption that we're going to use Vega-Embed with the promise feature which is not released yet. When testing, I linked Vega-Embed to the current master branch of Vega-Embed.

### CODE REVIEW

File: site/static/streaming.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ export function runStreamingExample(eleId: string) {
     }
   };
 
-  function cb(err: Error, res: {view: any, spec: any}) {
```
Comment (domoritz) [inline site/static/streaming.ts]: Let's move this code inside the `then` call.

File: site/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es6",
```
Comment (domoritz) [inline site/tsconfig.json]: Please don't change this! This means we generate es6 and this would break everyone who uses a browser that doesn't support es6.

File: site/usage/compile.md [hunk 3]
```diff
@@ -20,14 +20,6 @@ If you want access to the compiled Vega spec from a Javascript program, you can
 var vgSpec = vl.compile(vlSpec).spec;
 ```
 
-Then, you can continue to use the [Vega runtime](https://github.com/vega/vega/wiki/Runtime)'s `vg.parse.spec` method to render your Vega spec.
```
Comment (domoritz) [inline site/usage/compile.md]: What's wrong with this?

File: site/usage/embed.md [hunk 4]
```diff
@@ -9,22 +9,23 @@ permalink: /usage/embed.html
 
 The easiest way to use Vega-Lite on your own web page is with [Vega-Embed](https://github.com/vega/vega-embed), a library we built to make the process as painless as possible.
 
-To embed a Vega-Lite specification on your web page first load the required libraries (D3, Vega, Vega-Lite, **Vega-Embed**).
```
Comment (domoritz) [inline site/usage/embed.md]: Good catch.

File: site/usage/embed.md [hunk 5]
```diff
@@ -35,15 +36,45 @@ The next step is to create a DOM element that the visualization will be attached
...
 ```
 
 Vega-Embed automatically adds links to export an image, view the source, and open the specification in the online editor. These links can be individually disabled. For more information, read the [Vega-Embed documentation](https://github.com/vega/vega-embed).
+
+Here is the fianl html file in the easiest way to embed Vega-Lite:
+```html
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Embedding Vega-Lite</title>
```
Comment (domoritz) [inline site/usage/embed.md]: Above, we show how to use cdnjs, which I think should be the default for beginners.

File: site/tutorials/getting_started.md [hunk 6]
```diff
@@ -12,6 +12,14 @@ This tutorial will guide through the process of writing a visualization specific
 
 We suggest that you follow along the tutorial by building a visualization in the [online editor](https://vega.github.io/new-editor/?mode=vega-lite). Extend your specification in the editor as you read through this tutorial. If something does not work as expected, compare your specifications with ones inside this tutorial.
 
+## $schema
```
Comment (domoritz) [inline site/tutorials/getting_started.md]: out it in `

File: site/tutorials/getting_started.md [hunk 7]
```diff
@@ -12,6 +12,14 @@ This tutorial will guide through the process of writing a visualization specific
...
+
+Before diving into components illustrating visualizations, we want to introduce "$schema" property that defines the [JSON schema](http://json-schema.org/) of the visualization specification in a certain version of Vega-Lite. It accepts value as an URL of the schema.
```
Comment (domoritz) [inline site/tutorials/getting_started.md]: **the** $schema property

File: site/tutorials/streaming.md [hunk 8]
```diff
@@ -31,7 +31,7 @@ Putting it together, we have:
   view.change('table', changeSet).run();
 ```
 
-The view is given to us by the Vega [embed](https://github.com/vega/vega-embed) method, which help load Vega/Vega-Lite specs on your webpages. The callback argument to `embed` is invoked when embed is complete, with parameters: `error`, then an object containing the `view`, which is how we could access the `view.change` mentioned earlier, and `spec`, which is the compiled Vega spec from the original Vega-Lite spec given.
+The view is given to us by the Vega [embed](https://github.com/vega/vega-embed) method, which helps load Vega/Vega-Lite specs on your webpages. When `embed` completes successfully, `embed` returns a fulfilled [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) with an object containing the `view`, which is how we could access the `view.change` mentioned earlier, and `spec`, which is the compiled Vega spec from the original Vega-Lite spec given. When `embed` fails, it returns a rejected [promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) with an error object.
```
Comment (domoritz) [inline site/tutorials/streaming.md]: let's not call it the "Vega" embed method, because it is not part of vega.

File: site/usage/compile.md (hunk 3 above)
Comment (yhoonkim) [inline site/usage/compile.md]: 1) `vg.parse.spec` method is deprecated. (I think it is replaced by `vega.parse`.)
2) I thought we should promote to use Vega-Embed to render the compiled spec too.
3) I felt that this info is not related to *Compile* and not necessary to be here.
So I just decided to delete this incorrect info.

**Comment by yhoonkim** [comment]
Thanks @domoritz for reviews! I pushed a commit on this branch to reflect them.
(But I don't know why it is not shown in this thread...)

File: site/usage/compile.md (hunk 3 above)
Comment (domoritz) [inline site/usage/compile.md]: Makes sense. Thanks for the explanation.

**Comment by domoritz** [comment]
I pushed some small fixes. Can you also address https://github.com/vega/vega-lite/pull/2556/files#r123850983?
